### PR TITLE
Add dasel

### PR DESCRIPTION
**Pull request by maresb**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by maresb** [comment]
I know nothing about Go, I simply want to use this utility in my Conda environments.

I modeled this on various other feedstocks I found such as https://github.com/conda-forge/go-lab-feedstock/blob/main/recipe/meta.yaml 

@conda-forge/help-go ready for review

**Comment by maresb** [comment]
By the way, I've had a very difficult time trying to develop this locally, since my build requirements of `{{ compiler('go') }}` for some reason expand to:
```yaml
    - _libgcc_mutex 0.1 conda_forge
    - libgfortran4 7.5.0 h14aa051_20
    - libstdcxx-ng 12.1.0 ha89aaad_16
    - libgfortran-ng 7.5.0 h14aa051_20
    - libgomp 12.1.0 h8d9b700_16
    - _openmp_mutex 4.5 2_gnu
    - libgcc-ng 12.1.0 h8d9b700_16
    - go-cgo 1.11.3 hcb93f98_1002
    - go_linux-64 1.11.3 hfa3001e_1002
```
and 1.11.3 is a [very old version](https://github.com/golang/go/releases/tag/go1.11.3) of Go. This happens with both `mamba/boa` and `conda`, and I have tried in a fresh environment. Does anyone know what might be going on here?

File: recipes/dasel/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set name = "dasel" %}
+{% set author = "TomWright" %}
+{% set version = "1.24.3" %}
+{% set pkg_src = 'src/' + name %}
```
Comment (carterbox) [inline recipes/dasel/meta.yaml:4]: Please remove the jinja variables for all except version. Jinja variables are for things that need to change frequently. i.e. the version number, but the package name and directories should be relatively constant throughout the life of the recipe. Additionally, certain metadata is extracted by conda-forge from the about section without running Jinja.

**Review by carterbox — CHANGES_REQUESTED** [review]
Please [see this issue](https://github.com/conda-forge/conda-forge.github.io/issues/1052) about collecting the licenses of all the go dependencies of this package.

File: recipes/dasel/meta.yaml (hunk 1 above)
Comment (maresb) [reply recipes/dasel/meta.yaml]: Ok, thanks! This part was easy.

Now about the licenses........ I need to figure out what must be done there.

**Comment by maresb** [comment]
Anything I can do to move this along? It's still labeled "Awaiting author contribution" but hopefully my commits have addressed the issues.

**Comment by BastianZim** [comment]
You’ll actually need to ping the relevant team again to remove that label, just fyi. 
(Don‘t know much about go though, sorry :smile:) 
I’d recommend posting in gitter, that’s how most go recipes are merged.

**Pull request merged**